src/components/Drawer/CreateOrEditMonsterDrawer/index.tsx

```diff
@@ -0,0 +1,162 @@
+                            onClick={() => {
+                                editMonster ? edit(monster) : create(monster)
+                            }}
+                        >Submit</LoadingButton>
+                    </Grid>
+                    <Grid item>
+                        <Button onClick={onClose}><FontAwesomeIcon icon={faXmarkCircle}/></Button>
+                    </Grid>
+                </Grid>
+                <Grid container spacing={2} rowGap={2}>
+                    <Grid item xs={12} md={6}>
+                        <TextInput value={monster?.name} onChange={(value) => setMonster({ ...monster, name: String(value)})} placeholder='Name' />
+                    </Grid>
+                    <Grid item xs={12} md={3}>
+                        <SizeSelect value={monster.size} onChange={(value: CharacterSizes) => setMonster({ ...monster, size: value})} />
+                    </Grid>
+                    <Grid item xs={12} md={3}>
+                        <CharacterTypeSelect value={monster.type} onChange={(value: CharacterType) => setMonster({ ...monster, type: value})} />
+                    </Grid>
+                    <Grid item xs={12} md={4}>
+                        <TextInput value={monster?.characterImageURL} onChange={(value) => setMonster({ ...monster, characterImageURL: String(value)})} placeholder='Image URL' />
+                    </Grid>
+                    
+                    <Grid item xs={12} md={2}>
+                        <TextInput value={monster?.averageHitPoints} onChange={(value) => setMonster({ ...monster, averageHitPoints: Number(value)})} placeholder='Average Hit Points' number />
+                    </Grid>
+                    <Grid item xs={4} md={2}>
+                        <TextInput value={monster?.speed} onChange={(value) => setMonster({ ...monster, speed: Number(value)})} placeholder='Speed' number />
+                    </Grid>
+                    <Grid item xs={4} md={2}>
+                        <TextInput value={monster?.armorClass} onChange={(value) => setMonster({ ...monster, armorClass: Number(value)})} placeholder='Armor Class' number />
+                    </Grid>
+                    <Grid item xs={4} md={2}>
+                        <TextInput value={monster?.passivePerception} onChange={(value) => setMonster({ ...monster, passivePerception: Number(value)})} placeholder='Passive Percep' number />
+                    </Grid>
+                    <Grid item xs={4} md={1}>
+                        <TextInput value={monster?.abilityScores?.strength} onChange={(value) => {
+                            const scores = {...monster?.abilityScores, strength: Number(value)}
+                            setMonster({...monster, abilityScores: scores})
+                        }} placeholder={`Strength (${Math.floor((monster?.abilityScores?.strength - 10)/2) >= 0 ? "+" : ""}${Math.floor((monster?.abilityScores?.strength - 10)/2)})`} number />
```

Comment:
Random note - shoujld make these a util at some point lol. Its the same calc for every ability score type

Reply:
#truuuueeee